include/watchdog.h

```diff
@@ -55,62 +70,237 @@ struct wdt_config {
 	enum wdt_mode mode;
 	void (*interrupt_fn)(struct device *dev);
 };
+/**
+ * @}
+ */
+
 
-typedef void (*wdt_api_enable)(struct device *dev);
-typedef void (*wdt_api_disable)(struct device *dev);
-typedef int (*wdt_api_set_config)(struct device *dev,
-				  struct wdt_config *config);
-typedef void (*wdt_api_get_config)(struct device *dev,
-				   struct wdt_config *config);
-typedef void (*wdt_api_reload)(struct device *dev);
+/**
+ * @brief Pause watchdog timer when CPU is in sleep state.
+ */
+#define WDT_OPT_PAUSE_IN_SLEEP	BIT(0)
 
+/**
+ * @brief Pause watchdog timer when CPU is halted by the debugger.
+ */
+#define WDT_OPT_PAUSE_HALTED_BY_DBG	BIT(1)
+
+/**
+ * @name Watchdog Reset Behavior.
+ * If none of the reset flags is specified there will be no reset after timeout.
+ * @{
+ */
+/** CPU core reset */
+#define WDT_FLAG_RESET_CPU_CORE		BIT(0)
+/** Global system reset */
+#define WDT_FLAG_RESET_SOC		BIT(1)
+/**
+ * @}
+ */
+
+/**
+ * @brief Watchdog reset flag bit field mask.
+ */
+#define WDT_FLAG_RESET_MASK	(0x3)
+
+/**
+ * @brief Watchdog timeout window.
+ *
+ * Each installed timeout needs feeding in specified time window, otherwise
+ * the watchdog will trigger. If watchdog instance does not support window
+ * timeouts then min value must be equal to 0.
+ *
+ * @param min Lower limit of watchdog feed timeout in milliseconds.
+ * @param max Upper limit of watchdog feed timeout in milliseconds.
+ *
+ * @note If specified values can not be precisely set they are always
+ *	 rounded up.
+ */
+struct wdt_window {
+	u32_t min;
+	u32_t max;
+};
+
+/** Watchdog callback. */
+typedef void (*wdt_callback_t)(struct device *dev, void *handle);
+
+/**
+ * @brief Watchdog timeout configuration struct.
+ *
+ * @param window Timing parameters of watchdog timeout.
+ * @param handle Unique handle used to feed specific watchdog timeout.
+ *		 It is allowed to use the same handle for multistage timeouts.
+ *		 This handle is filled by the user, it will be supplied to
+ *		 callback. Good example is handle pointing to the id of the
+ *		 thread that is resposible for feeding corresponding timeout.
+ * @param callback Timeout callback. Passing NULL means that no callback
+ *		   will be run.
+ * @param next Pointer to the next timeout configuration. It is used for
```

Comment:
Better would be:

     Pointer to the next timeout configuration. This pointer is used for
     watchdogs with staged timeouts functionality. Value must be NULL
     for single stage timeouts.